services/scan/src/precinct_scanner_state_machine.ts

```diff
@@ -0,0 +1,854 @@
+import {
+  ClientDisconnectedError,
+  createClient,
+  DEFAULT_CONFIG,
+  PaperStatus,
+  ScannerClient,
+  ScannerError,
+} from '@votingworks/plustek-sdk';
+import { v4 as uuid } from 'uuid';
+import {
+  AdjudicationReason,
+  AdjudicationReasonInfo,
+  err,
+  Id,
+  ok,
+  PageInterpretationWithFiles,
+  Result,
+} from '@votingworks/types';
+import { assert } from '@votingworks/utils';
+import { switchMap, timer } from 'rxjs';
+import {
+  assign as xassign,
+  Assigner,
+  createMachine,
+  interpret,
+  InvokeConfig,
+  PropertyAssigner,
+  send,
+} from 'xstate';
+import { Scan } from '@votingworks/api';
+import makeDebug from 'debug';
+import {
+  deleteInterpretedSheet,
+  SimpleInterpreter,
+  storeInterpretedSheet,
+} from './simple_interpreter';
+import { SheetOf } from './types';
+import { Store } from './store';
+
+const debug = makeDebug('scan:precinct-scanner');
+const debugPaperStatus = debug.extend('paper-status');
+const debugEvents = debug.extend('events');
+
+// Temporary mode to control whether we interpreting ballots or not
+export type InterpretationMode = 'interpret' | 'skip';
+
+class PrecinctScannerError extends Error {
+  // eslint-disable-next-line vx/gts-no-public-class-fields
+  constructor(public type: Scan.PrecinctScannerErrorType, message?: string) {
+    super(message ?? type);
+  }
+}
+
+export interface Context {
+  store?: Store;
+  interpreter?: SimpleInterpreter;
+  client?: ScannerClient;
+  scannedSheet?: SheetOf<string>;
+  interpretation?: InterpretationResultEvent;
+  error?: Error;
+  interpretationMode: InterpretationMode;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function assign(arg: Assigner<Context, any> | PropertyAssigner<Context, any>) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return xassign<Context, any>(arg);
+}
+
+type ConfigurationEvent =
+  | { type: 'CONFIGURE'; store: Store; interpreter: SimpleInterpreter }
+  | { type: 'UNCONFIGURE' };
+
+type ScannerStatusEvent =
+  | { type: 'SCANNER_NO_PAPER' }
+  | { type: 'SCANNER_READY_TO_SCAN' }
+  | { type: 'SCANNER_READY_TO_EJECT' }
+  | { type: 'SCANNER_BOTH_SIDES_HAVE_PAPER' }
+  | { type: 'SCANNER_JAM' }
+  | { type: 'SCANNER_DISCONNECTED' };
+
+interface Interpretation {
+  sheetId: Id;
+  sheet: SheetOf<PageInterpretationWithFiles>;
+}
+
+export type InterpretationResultEvent = Scan.InterpretationResult & {
+  interpretation: Interpretation;
+};
+
+type CommandEvent =
+  | { type: 'SCAN' }
+  | { type: 'ACCEPT' }
+  | { type: 'RETURN' }
+  | { type: 'START_OVER' }
```

Comment:
nit: `START_OVER` is not clear as a command compared to others. Feels almost like a catch-all. Maybe it isn't, just commenting on naming.

Reply:
Good to know. Do you have any ideas for a clearer name? It's used for both going back to "no_paper" from "accepted" and "calibrated" so it is doing double duty a bit.

Maybe something like "WAIT_FOR_PAPER" or "RESET_TO_NO_PAPER"?

Reply:
`WAIT_FOR_PAPER` is great.

Reply:
8ca47de57